### PR TITLE
Melodic compatibility

### PR DESCRIPTION
**Pull request by simonschmeisser**
this series of patches fixes compilation on melodic while keeping compatibility to at least kinetic. It is mostly clean and obvious fixes except for the last commit where maybe checking the MoveIt! version would be preferable

### CODE REVIEW

File: stomp_moveit/src/utils/kinematics.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <kdl_parser/kdl_parser.hpp>
 #include <eigen_conversions/eigen_kdl.h>
 #include <math.h>
+#include <random>
```
Comment (jrgnicho) [inline stomp_moveit/src/utils/kinematics.cpp:31]: You mentioned adding this in another PR.  I'm not sure why it is needed now, could you briefly explain?

File: constrained_ik/src/moveit_interface/joint_interpolation_planner.cpp [hunk 2]
```diff
@@ -63,7 +63,11 @@ namespace constrained_ik
     const robot_model::JointModelGroup *group_model = rob_model->getJointModelGroup(request_.group_name);
     std::vector<std::string> joint_names = group_model->getActiveJointModelNames();
     std::vector<std::string> link_names = group_model->getLinkModelNames();
-    Eigen::Affine3d goal_pose;
+    #if ROS_VERSION_MINIMUM(1, 14, 0) //melodic
```
Comment (jrgnicho) [inline constrained_ik/src/moveit_interface/joint_interpolation_planner.cpp:66]: I feel that is change is part of a different larger issue and not necessarily melodic migration related

File: industrial_collision_detection/package.xml [hunk 3]
```diff
@@ -15,13 +15,13 @@
...
   <build_depend>pluginlib</build_depend>
-  <build_depend>pcl_ros</build_depend>
+  <build_depend>libfcl-dev</build_depend>
 
   <run_depend>roscpp</run_depend>
   <run_depend>moveit_core</run_depend>
   <run_depend>cmake_modules</run_depend>
   <run_depend>pluginlib</run_depend>
-  <run_depend>pcl_ros</run_depend>
+  <run_depend>libfcl-dev</run_depend>
```
Comment (jrgnicho) [inline industrial_collision_detection/package.xml:24]: I can't think of a good reason why pcl_ros made its way into here.  Glad you caught this.

File: industrial_collision_detection/src/collision_detection/collision_common.cpp [hunk 4]
```diff
@@ -78,7 +78,7 @@ namespace collision_detection
         {
           always_allow_collision = true;
           if (!cdata->req->verbose)
-            logDebug("Collision between '%s' and '%s' is always allowed. No contacts are computed.",
+            CONSOLE_BRIDGE_logDebug("Collision between '%s' and '%s' is always allowed. No contacts are computed.",
```
Comment (jrgnicho) [inline industrial_collision_detection/src/collision_detection/collision_common.cpp:81]: I prefer using the ros console macros but it's fine by me if it stays like this as this package is currently deprecated.

**Comment by jrgnicho** [comment]
@simonschmeisser thanks for submitting this.  I have yet to test the build in a melodic docker container  so I'll be doing that and get back with you.

File: stomp_moveit/src/utils/kinematics.cpp (hunk 1 above)
Comment (simonschmeisser) [reply stomp_moveit/src/utils/kinematics.cpp:31]: I guess it was included indirectly in kinetic and that somehow changed in melodic. I get compiler errors about
`
  std::random_device rd;
  std::mt19937 gen(rd());
`
in lines 458ff

File: constrained_ik/src/moveit_interface/joint_interpolation_planner.cpp (hunk 2 above)
Comment (simonschmeisser) [reply constrained_ik/src/moveit_interface/joint_interpolation_planner.cpp:66]: https://github.com/ros-planning/moveit/pull/1096

File: industrial_collision_detection/src/collision_detection/collision_common.cpp (hunk 4 above)
Comment (simonschmeisser) [reply industrial_collision_detection/src/collision_detection/collision_common.cpp:81]: I just went for the least invasive way ...

**Comment by jrgnicho** [comment]
This is minor but I got some compile warnings on a deprecated header

```
In file included from /root/industrial_moveit_ws/src/industrial_moveit/stomp_moveit/src/stomp_planner.cpp:29:0:
/opt/ros/melodic/include/class_loader/class_loader.h:36:2: warning: #warning Including header <class_loader/class_loader.h> is deprecated, include <class_loader/class_loader.hpp> instead. [-Wcpp]
 #warning Including header <class_loader/class_loader.h> is deprecated, \
  ^~~~~~~
In file included from /root/industrial_moveit_ws/src/industrial_moveit/stomp_moveit/src/stomp_planner_manager.cpp:26:0:
/opt/ros/melodic/include/class_loader/class_loader.h:36:2: warning: #warning Including header <class_loader/class_loader.h> is deprecated, include <class_loader/class_loader.hpp> instead. [-Wcpp]
 #warning Including header <class_loader/class_loader.h> is deprecated, \

.
.
.

                 from /root/industrial_moveit_ws/src/industrial_moveit/constrained_ik/src/moveit_interface/constrained_ik_planner_plugin.cpp:26:
/opt/ros/melodic/include/class_loader/class_loader.h:36:2: warning: #warning Including header <class_loader/class_loader.h> is deprecated, include <class_loader/class_loader.hpp> instead. [-Wcpp]
 #warning Including header <class_loader/class_loader.h> is deprecated, \

```

**Comment by jrgnicho** [comment]
Could you rebase against the latest in order to have the fix to the constrained_ik unit test and have it pass the travis ci build?

File: constrained_ik/src/moveit_interface/joint_interpolation_planner.cpp (hunk 2 above)
Comment (jrgnicho) [reply constrained_ik/src/moveit_interface/joint_interpolation_planner.cpp:66]: There are lots of places in industrial_moveit where `Eigen::Affine3d` types are used so I think the proper way to tackle this is to have perhaps another PR that replaces all of those with the `Eigen::Isometry3d ` type.

File: stomp_moveit/src/utils/kinematics.cpp (hunk 1 above)
Comment (jrgnicho) [reply stomp_moveit/src/utils/kinematics.cpp:31]: You are right about the error, I just verified this.

**Comment by simonschmeisser** [comment]
sure, I'll do that tomorrow when I'm back in office

about the warnings I need to check if that would be backwards compatible

**Comment by simonschmeisser** [comment]
ok, travis succeeds now

**Comment by simonschmeisser** [comment]
the class_loader change should be backwards compatible as well, lets see what travis says

**Comment by jrgnicho** [comment]
@simonschmeisser thanks for the contribution.

**Pull request merged**